src/Libraries/Microsoft.Extensions.AI.Abstractions/Tools/HostedMcpServerTool.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Represents a hosted MCP server tool that can be specified to an AI service.
+/// </summary>
+public class HostedMcpServerTool : AITool
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="HostedMcpServerTool"/> class.
+    /// </summary>
+    /// <param name="serverName">The name of the remote MCP server.</param>
+    /// <param name="url">The URL of the remote MCP server.</param>
+    public HostedMcpServerTool(string serverName, Uri url)
+    {
+        ServerName = Throw.IfNullOrWhitespace(serverName);
+        Url = Throw.IfNull(url);
+    }
+
+    /// <summary>
+    /// Gets the name of the remote MCP server that is used to identify it.
+    /// </summary>
+    public string ServerName { get; }
+
+    /// <summary>
+    /// Gets the URL of the remote MCP server.
+    /// </summary>
+    public Uri Url { get; }
+
+    /// <summary>
+    /// Gets or sets the description of the remote MCP server, used to provide more context to the AI service.
+    /// </summary>
+    public string? ServerDescription { get; set; }
+
+    /// <summary>
+    /// Gets or sets the list of tools allowed to be used by the AI service.
+    /// </summary>
```

Comment:
The comment should specify behavior when the list is null. Presumably that's different from empty, where null means everything is allowed but empty means nothing is allowed?